### DISCUSSION TITLE
IfcConvert no materials exported in OBJ since 0.6.0b0

### DISCUSSION THREAD

**Discussion by ben-fly2success** [IfcOpenShell/IfcOpenShell#4182] [Q&A] [answered]
Dear community,

I tried various versions of IfcConvert and the only one that export correctly materials and colors from IFC to OBJ (opened in Cloudcompare for exemple) is the 0.6.0b0 OCC 7.3.0. All version after this one export MTL but when opening the OBJ it's not linked.

Did you notice the same issue ? Is it a cloudcompare issue or IfcConvert ?

Best regards,
Benjamin

**Comment by aothms** [comment]
If something fundamental like that would be broken we would have heard earlier, so something specific to your situation must be affecting this. For starters what are first couple lines of the working and non-working OBJ files where the MTL is referenced? Is there a difference?

**Reply by ben-fly2success** [reply]
I didn't know OBJ files are text file. You're right the mtllib line is missing from the 0.7.0, maybe people didn't report it as they knew only adding this line makes it work ?

# File generated by IfcOpenShell v0.7.0-f0e03c79d
s 1
v 1646855.71440567 8181471.87870847 29.7799998474127
v 1646855.71440567 8181471.87870847 32.3799998474128

# File generated by IfcOpenShell 0.6.0b0
mtllib ARC_BAT_PKG_detache.mtl
g product-57cab55f-0535-4d25-9519-5216c264f275-body
s 1
v 1646869.51768656 8181430.88563091 29.7799998474127
v 1646869.51768656 8181430.88563091 32.3799998474128
v 1646866.97222367 8181427.05958188 29.7799998474127

I used the Anaconda build on windows and build on MacOs, both have the same issue.

**Reply by aothms** [reply, marked as answer]
> If something fundamental like that would be broken we would have heard earlier

Wow, famous last words. There was a bug, but it seems a lot of OBJ importers try to find a default MTL file based on the OBJ filename even if it's not specified, which explains why it wasn't observed earlier, e.g:

https://github.com/blender/blender-addons/blob/cb2f133712599040b2f5ffaa1f9ff2e3a1e55e73/io_scene_obj/import_obj.py#L206-L207

Thanks for reporting and posting the content of the files.
